Fix inlining for apply operators

garyb opened the pull request:
Resolves #2298

paf31 commented on src/Language/PureScript/CodeGen/JS/Optimizer/Inliner.hs:
```diff
@@ -107,6 +107,7 @@ inlineOperator (m, op) f = everywhereOnJS convert
   convert (JSApp _ (JSApp _ op' [x]) [y]) | isOp op' = f x y
   convert other = other
   isOp (JSIndexer _ (JSStringLiteral _ op') (JSVar _ m')) = m == m' && op == op'
```
Do we still need the `JSIndexer` case?

paf31 commented:
👍 LGTM, thanks!

garyb commented:
You're right, we don't need the other case, and in fact there are a bunch of other things we don't need now too, as they were for the pre-0.9 release before we made breaking changes. I'll update.

garyb commented:
Looks like they had already mostly been sorted out actually. I moved these into what seemed a more appropriate place though, and removed the redundant cases (both in terms of inlining-via-indexer and inlining the operators rather than the named functions).

paf31 commented:
👍 Thanks!
